bokeh/core/has_props.py

```diff
@@ -463,7 +464,16 @@ def lookup(cls, name):
             PropertyDescriptor : descriptor for property named ``name``
 
         '''
-        return getattr(cls, name)
+        attr = cls.lookup_descriptor(name)
+        if attr is not None:
+            return attr
+        else:
```

Comment:
`else` is unnecessary here after early return. 

That said, it's weird to add a new function named `lookup_descriptor` when the explicit purpose of the `lookup` function is to look up descriptors. I would personally suggest keeping only `lookup` but adding a boolean flag to indicate whether a missing name raises or returns `None`. Alternatively `lookup_desriptor ` should get a different name that represents the actual important difference (i.e. that there is no exception), e.g. `lookup_soft` or `lookup_safe` or something.

Reply:
This is a proof of concept whether this approach works at all. Whatever is the naming or whether those two methods are merged together, is a secondary concern right now.

Reply:
I we want to wait on good naming, we should make it private then, at least: `_lookup_descriptor`